### PR TITLE
Checkout migration docs improvement

### PR DESCRIPTION
**Pull request by bgambocjaviniar**
closes https://jira.tools.sap/browse/CXSPA-1509

### CODE REVIEW

File: _pages/home/updating-to-version-5/checkout-libraries-release-notes.md [hunk 1]
```diff
@@ -223,6 +223,42 @@ Spartacus schematics are designed to assist you in your migration and to help ma
 
 Take a look at the [5.0 TypeScript Changes](https://github.com/SAP/spartacus/blob/84e6b462db1f5dca5c3145c5bc535fc5e2b52fe2/docs/migration/5_0-generated-typescript-changes-doc.md) to get a high-level overview of what has changed.
 
+### Configuring checkout that is already configured to load from the old checkout entrypoint '@spartacus/checkout/root' either eagerly or lazily.
```
Comment (znikola) [inline _pages/home/updating-to-version-5/checkout-libraries-release-notes.md:226]:
```suggestion
### Migrating the old checkout configuration
```
Maybe?

File: _pages/home/updating-to-version-5/checkout-libraries-release-notes.md [hunk 2]
```diff
@@ -223,6 +223,42 @@ Spartacus schematics are designed to assist you in your migration and to help ma
...
+To re-iterate, if you are using Spartacus 4.0, the checkout library only contains a single entrypoint (`@spartacus/checkout`), which contains all checkout flows. On the other hand, in Spartacus 5.0, the library has been further decoupled into different business logic features, with the intention of having a smaller bundle size by splitting it into three different entrypoints instead (`@spartacus/checkout/base/root`, `@spartacus/checkout/b2b/root`, or `@spartacus/checkout/scheduled-replenishment/root`).
+
+If your application is already configured to load from `@spartacus/checkout/root`, you would need to update the reference that imports from it by choosing from either of the three different entrypoints as mentioned earlier.
+
+However, if there were added configurations that adds some customized logic to checkout, you might consider creating a wrapper module, which allows additional extensions/functionality to work with your customized checkout module.
+
+The following is an example on how to approach the creation of a wrapper module, which will help with lazy loading the feature:
+
+```ts
+// wrapper module file
```
Comment (znikola) [inline _pages/home/updating-to-version-5/checkout-libraries-release-notes.md:237]: Let's keep it consistent with how our shell app and schematics are generating files, and name it: `checkout-wrapper.module.ts`

File: _pages/home/updating-to-version-5/checkout-libraries-release-notes.md [hunk 3]
```diff
@@ -223,6 +223,42 @@ Spartacus schematics are designed to assist you in your migration and to help ma
 
 Take a look at the [5.0 TypeScript Changes](https://github.com/SAP/spartacus/blob/84e6b462db1f5dca5c3145c5bc535fc5e2b52fe2/docs/migration/5_0-generated-typescript-changes-doc.md) to get a high-level overview of what has changed.
 
+### Configuring checkout that is already configured to load from the old checkout entrypoint '@spartacus/checkout/root' either eagerly or lazily.
+
+To re-iterate, if you are using Spartacus 4.0, the checkout library only contains a single entrypoint (`@spartacus/checkout`), which contains all checkout flows. On the other hand, in Spartacus 5.0, the library has been further decoupled into different business logic features, with the intention of having a smaller bundle size by splitting it into three different entrypoints instead (`@spartacus/checkout/base/root`, `@spartacus/checkout/b2b/root`, or `@spartacus/checkout/scheduled-replenishment/root`).
```
Comment (znikola) [inline _pages/home/updating-to-version-5/checkout-libraries-release-notes.md:228]: Let's just use the main entry point path, and omit the `/root`.

File: _pages/home/updating-to-version-5/checkout-libraries-release-notes.md [hunk 4]
```diff
@@ -223,6 +223,42 @@ Spartacus schematics are designed to assist you in your migration and to help ma
...
+
+If your application is already configured to load from `@spartacus/checkout/root`, you would need to update the reference that imports from it by choosing from either of the three different entrypoints as mentioned earlier.
```
Comment (znikola) [inline _pages/home/updating-to-version-5/checkout-libraries-release-notes.md:230]: Not really. B2B and Repl both require Base. 
Moreover, Repl requires B2B.

File: _pages/home/updating-to-version-5/checkout-libraries-release-notes.md [hunk 5]
```diff
@@ -223,6 +223,42 @@ Spartacus schematics are designed to assist you in your migration and to help ma
...
+
+However, if there were added configurations that adds some customized logic to checkout, you might consider creating a wrapper module, which allows additional extensions/functionality to work with your customized checkout module.
```
Comment (znikola) [inline _pages/home/updating-to-version-5/checkout-libraries-release-notes.md:232]: Wrapper module is the recommended way even if there are no customizations, and customers just want to LL B2B and / or Repl checkout.
Wrapper modules are not required for the Base checkout.

File: _pages/home/updating-to-version-5/checkout-libraries-release-notes.md [hunk 6]
```diff
@@ -223,6 +223,42 @@ Spartacus schematics are designed to assist you in your migration and to help ma
...
+The following is an example on how to approach the creation of a wrapper module, which will help with lazy loading the feature:
+
+```ts
+// wrapper module file
+// Adding scheduled replenishment to your existing customized checkout feature
+import { NgModule } from '@angular/core';
+import { CheckoutScheduledReplenishmentModule } from '@spartacus/checkout/scheduled-replenishment';
+import { CustomizedCheckoutModule } from './some/path/of/your/customized/checkout/module';
+@NgModule({
+  imports: [CustomizedCheckoutModule, CheckoutScheduledReplenishmentModule],
```
Comment (znikola) [inline _pages/home/updating-to-version-5/checkout-libraries-release-notes.md:243]: It's worth mentioning that Repl requires B2B _and_ Base checkout.

```ts
@NgModule({
  imports: [CheckoutModule, CheckoutB2BModule, CheckoutScheduledReplenishmentModule, CustomizedCheckoutModule],
})
export class CheckoutWrapperModule {}
```

File: _pages/home/updating-to-version-5/checkout-libraries-release-notes.md [hunk 7]
```diff
@@ -223,6 +223,42 @@ Spartacus schematics are designed to assist you in your migration and to help ma
...
+})
+export class SomeCustomizedCheckoutWrapperModule {}
+```
+
+```ts
+// Within your CustomizedFeatureModule, you use the newly created wrapper module that contains your customized checkout + scheduled replenishment
```
Comment (znikola) [inline _pages/home/updating-to-version-5/checkout-libraries-release-notes.md:249]: Let's add the file name, to make it consistent with the above snippet:
`checkout-feature.module.ts`

File: _pages/home/updating-to-version-5/checkout-libraries-release-notes.md [hunk 8]
```diff
@@ -223,6 +223,42 @@ Spartacus schematics are designed to assist you in your migration and to help ma
...
+    [CHECKOUT_FEATURE]: {
+      module: () =>
+        import('./some-customized-checkout-wrapper.module').then(
+          (m) => m.SomeCustomizedCheckoutWrapperModule
+        ),
+    },
+  },
+}),
+```
+
```
Comment (znikola) [inline _pages/home/updating-to-version-5/checkout-libraries-release-notes.md:261]: We should leave a note saying the above is the LL setup.
Maybe we can also show an eager setup

**Review by znikola — CHANGES_REQUESTED** [review]

File: _pages/home/updating-to-version-5/checkout-libraries-release-notes.md (hunk 8 above)
Comment (bgambocjaviniar) [reply _pages/home/updating-to-version-5/checkout-libraries-release-notes.md:261]: I thought the eager setup would be self explanatory to customers when we mention dynamic imports, but will add something for this

File: _pages/home/updating-to-version-5/checkout-libraries-release-notes.md (hunk 1 above)
Comment (bgambocjaviniar) [reply _pages/home/updating-to-version-5/checkout-libraries-release-notes.md:226]: that's also fine to me :d

File: _pages/home/updating-to-version-5/checkout-libraries-release-notes.md (hunk 4 above)
Comment (bgambocjaviniar) [reply _pages/home/updating-to-version-5/checkout-libraries-release-notes.md:230]: gotcha, I forgot 😓

File: _pages/home/updating-to-version-5/checkout-libraries-release-notes.md (hunk 8 above)
Comment (znikola) [reply _pages/home/updating-to-version-5/checkout-libraries-release-notes.md:328]: Works for me, I'm also thinking it is self explanatory.

File: _pages/home/updating-to-version-5/checkout-libraries-release-notes.md [hunk 9]
```diff
@@ -223,42 +223,82 @@ Spartacus schematics are designed to assist you in your migration and to help ma
...
 
-If your application is already configured to load from `@spartacus/checkout/root`, you would need to update the reference that imports from it by choosing from either of the three different entrypoints as mentioned earlier.
+If your application is already configured to load from `@spartacus/checkout`, you would need to update the reference that imports from it by changing it to `@spartacus/checkout/base`.
 
-However, if there were added configurations that adds some customized logic to checkout, you might consider creating a wrapper module, which allows additional extensions/functionality to work with your customized checkout module.
+However, if there were added configurations that adds some customized logic to checkout, you might consider creating a wrapper module, which allows additional extensions/functionality to work with your customized checkout module. Creating a wrapper module is the recommended way even if there are no additional customizations.
```
Comment (znikola) [inline _pages/home/updating-to-version-5/checkout-libraries-release-notes.md:232]: I think I wasn't clear [here](https://github.com/SAP/spartacus-docs/pull/1505#discussion_r996851047), sorry for that.
What I meant was: if customers are using B2B and / or Repl, it is recommended to use the wrapper modules (regardless of their customizations). 
If customers want only Base Checkout, by default we don't generate a wrapper module (but as soon as customers want to add some customization, they will have to create it).

That being said, I think we don't need to have too many examples here. Only the "big" example (with `imports: [CheckoutModule, CheckoutB2BModule, CheckoutScheduledReplenishmentModule, CustomizedCheckoutModule]`) should be sufficient IMO.

File: _pages/home/updating-to-version-5/checkout-libraries-release-notes.md (hunk 8 above)
Comment (znikola) [reply _pages/home/updating-to-version-5/checkout-libraries-release-notes.md:328]: Yeah, I was thinking the same actually. It's up to you, I don't mind if we don't mention it.

**Review by gladius-mtl — APPROVED** [review]

**Pull request merged**